src/main/java/duke/DukeController.java

```diff
@@ -0,0 +1,245 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+public class DukeController {
+    protected ArrayList<Task> taskList = new ArrayList<Task>(0);
+    private final String GREETING_PART_ONE = "Hello! I'm Duke!";
+    private final String GREETING_PART_TWO = "What can I do for you?";
+    private final String GOODBYE = "Bye. Hope to see you again soon!";
+    private final String NUMBER_OF_TASKS_MESSAGE = "You now have %d tasks";
+    private final String TASK_ADDED_SUCCESSFULLY_MESSAGE = "Got it. I've added this task:";
+    private final String EMPTY_TASK_LIST_MESSAGE = "You have no tasks in your list currently!";
+    private final String LISTING_TASKS_MESSAGE = "Here are the tasks in your list:";
+    private final String UPDATE_TASK_SUCCESS = "Nice! I've %sed this task:";
+    private final String ILLEGAL_INDEX_ERROR = ":( %d is out of bounds of the task list";
+    private final String UPDATE_TASK_FAILURE = ":( This task is already %sed";
+    private final String DELETE_TASK_SUCCESS = "Nice! I've deleted this task:";
+    private final String SAVE_FILE_SUCCESS = "Nice! I've saved all your tasks!";
+    private final String LOAD_FILE_SUCCESS = "Nice! I've loaded your tasks from previous sessions!";
+    private final String CREATE_DIRECTORY_FAILED = "Please create an empty directory named %s in the folder containing ur .jar file!";
+    private final int TASK_TYPE = 0;
+    private final int TASK_STATUS = 1;
+    private final int TASK_DESCRIPTION = 2;
+    private final int TASK_TIMING = 3;
+    private final String DATA_DIRECTORY = "data";
```

Comment:
[Code Quality] Consider isolating these UI-related methods to a dedicated UI class.